ethereum_history_api/circuits/lib/src/verifiers/receipt_test.nr

```diff
@@ -1,43 +1,65 @@
-use crate::verifiers::receipt::assert_tx_idx_equals;
+use dep::proof::utils::memcpy;
+use crate::fixtures::mainnet::{
+    cancun::small_block::{receipt::{encoded_receipt, receipt, tx_type}, header::number as block_number},
+    homestead::fork::{
+    receipt::{encoded_receipt as pre_byzantium_encoded_receipt, tx_type as legacy_tx_type},
+    header::number as pre_byzantium_block_number
+}
+};
+use crate::misc::arrays::{alter_array, sub_array_equals_up_to_length};
+use crate::verifiers::receipt::{
+    BYZANTIUM_BLOCK_NUM, assert_receipt_equals, split_receipt_into_tx_type_and_receipt_rlp,
+    assert_receipt_rlp_equal, MAX_RECEIPT_RLP_LENGTH, MAX_ENCODED_RECEIPT_LENGTH
+};
 
 #[test]
-fn test_assert_tx_idx_equals_zero() {
-    let tx_idx = 0x0;
-    let key = [0x0, 0x0, 0x0, 0x0, 0x8, 0x0];
-    assert_tx_idx_equals(key, tx_idx);
+fn test_split_receipt_into_tx_type_and_receipt_rlp() {
+    let (tx_type, receipt_rlp) = split_receipt_into_tx_type_and_receipt_rlp(tx_type, encoded_receipt);
+    assert(tx_type == encoded_receipt[0], "Incorrect tx type returned");
+    assert(receipt_rlp.len() == encoded_receipt.len() - 1, "Incorrect receipt RLP length returned");
+    assert(receipt_rlp[0] == encoded_receipt[1], "Incorrect receipt RLP returned");
 }
 
 #[test]
-fn test_assert_tx_idx_equals_odd() {
-    let tx_idx = 0xf;
-    let key = [0x0, 0x0, 0x0, 0x0, 0x0, 0xf];
-    assert_tx_idx_equals(key, tx_idx);
+fn test_split_receipt_into_tx_type_and_receipt_rlp_legacy_tx() {
+    let (tx_type, receipt_rlp) = split_receipt_into_tx_type_and_receipt_rlp(legacy_tx_type, pre_byzantium_encoded_receipt);
+    assert(tx_type == legacy_tx_type, "Incorrect tx type returned");
+    assert(receipt_rlp.len() == encoded_receipt.len() - 1, "Incorrect receipt RLP length returned");
+    assert(receipt_rlp[0] == pre_byzantium_encoded_receipt[0], "Incorrect receipt RLP returned");
 }
 
-#[test]
-fn test_assert_tx_idx_equals() {
-    let tx_idx = 0xabcd;
-    let key = [0x8, 0x2, 0xa, 0xb, 0xc, 0xd];
-    assert_tx_idx_equals(key, tx_idx);
+#[test(should_fail_with="Not 0x00 at the end of legacy receipt RLP")]
+fn test_split_receipt_into_tx_type_and_receipt_rlp_wrong_legacy_rlp() {
+    let mut wrong_legacy_rlp = [0; MAX_ENCODED_RECEIPT_LENGTH];
+    memcpy(&mut wrong_legacy_rlp, pre_byzantium_encoded_receipt, 0);
+
+    let not_zero_byte = 0x01;
+    wrong_legacy_rlp[MAX_ENCODED_RECEIPT_LENGTH - 1] = not_zero_byte;
+
+    let(tx_type, receipt_rlp) = split_receipt_into_tx_type_and_receipt_rlp(legacy_tx_type, wrong_legacy_rlp);
```

Comment:
Missing space after let, also do we have to assign return values from split_receipt_... to variables? If yes, let's use let (\_, \_) = split_receipt_...